### ISSUE TITLE
When mixing fork branch names: Failed to get unambiguous name for "main"

### ISSUE THREAD

**Issue by darkfeline** [radian-software/straight.el#922]
<details><summary>Test Case</summary>

```emacs-lisp
(straight-bug-report
  :post-bootstrap 
  (straight-use-package
   '(org :inherit t :fork
         (:host github :repo "darkfeline/org-mode" :branch "master")))
  (straight-merge-package "org" t)
  :preserve t)
```
</details>

- Test run at: `2022-01-30 17:36:48`
- system-type: `gnu/linux`
- straight-version: `prerelease (HEAD -> master, origin/master, origin/HEAD) af5437f 2021-09-27`
- emacs-version: `GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.27, cairo version 1.17.4)
 of 2021-03-26`

<details><summary>Output</summary>

```emacs-lisp
Bootstrapping straight.el...
Bootstrapping straight.el...done
Rebuilding all packages due to build cache schema change
Looking for gnu-elpa-mirror recipe → Cloning melpa...
Looking for gnu-elpa-mirror recipe → Cloning melpa...done
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe → Cloning el-get...
Looking for emacsmirror-mirror recipe → Cloning el-get...done
Looking for straight recipe → Cloning emacsmirror-mirror...
Looking for straight recipe → Cloning emacsmirror-mirror...done
Building straight...
Building straight...done

Test run with version: prerelease (HEAD -> develop, origin/develop) 3b0f1a1 2022-01-08
Cloning org...
Cloning org...done
Building org...
Building org...done

Packages:
"org-elpa"                n/a                  n/a
"melpa"                   n/a                  master 4ae4dee3 2022-01-30
"gnu-elpa-mirror"         n/a                  master 9a086d6 2022-01-30
"el-get"                  melpa                master 93533097 2021-12-24
"emacsmirror-mirror"      n/a                  master 4b116fa 2022-01-25
"straight"                n/a                  develop 3b0f1a1 2022-01-08
"org"                     org-elpa             master 2bae58361 2022-01-30

Error: (error "Cannot display prompt in batch mode: In repository \"org\", failed to get status:

  Failed to get unambiguous name for \"main\"
  fatal: Needed a single revision")

```
</details>

**Comment by raxod502** [comment]
OK, this looks like the same issue as https://github.com/radian-software/prescient.el/issues/122, I'm going to see if I can take a look and resolve.

**Comment by raxod502** [comment]
Think the way to resolve this is to ensure that `straight-vc-git--ensure-local` creates a local copy of the upstream branch if it is absent.

**Comment by raxod502** [comment]
Should be fixed by https://github.com/radian-software/straight.el/commit/07cedf0c2b1a649cf3e378e4ba4b9943441c23d6. I now get the following expected output from your bug report case:

<details><summary>Output</summary>

```emacs-lisp
Bootstrapping straight.el...
Bootstrapping straight.el...done
Looking for gnu-elpa-mirror recipe → Cloning melpa...
Looking for gnu-elpa-mirror recipe → Cloning melpa...done
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe → Cloning el-get...
Looking for emacsmirror-mirror recipe → Cloning el-get...done
Looking for straight recipe → Cloning emacsmirror-mirror...
Looking for straight recipe → Cloning emacsmirror-mirror...done
Building straight...
Building straight...done

Test run with version: prerelease (HEAD -> develop, origin/develop) 07cedf0 2022-06-03
Cloning org...
Cloning org...done
Building org...
Building org...done

Packages:
"straight"                n/a                  develop 07cedf0 2022-06-03
"org-elpa"                n/a                  n/a
"melpa"                   n/a                  master 5be08102 2022-06-03
"gnu-elpa-mirror"         n/a                  master 808923d 2022-05-25
"el-get"                  melpa                master bf3dba44 2022-06-02
"emacsmirror-mirror"      n/a                  master 2e1ceba 2022-06-01
"org"                     org-elpa             master fdc052bd7 2022-05-26

Error: (error "Cannot display prompt in batch mode: In repository \"org\", HEAD on default branch \"master\" has diverged from \"origin/main\"")

```
</details>

Let me know if it works for you now!

**Comment by darkfeline** [comment]
Yes, it works for me on `07cedf0`.  (Although I confess I now use a workaround workflow for updating my packages. If/when I try using a `straight-merge-package` based workflow again and run into any other issues, I'll file a new bug.)

**Comment by alexkehayias** [comment]
This change caused `straight-thaw-versions` to fail for me with `wrong-type-argument arrayp nil` on each package being processed. 

Reverting to commit `d2940b812fcf87f7b42c0f56ad25a8b1f4cd2389` fixes it (emacs 28.1, macOS, straight develop branch).

**Comment by raxod502** [comment]
Also reported in https://github.com/radian-software/straight.el/issues/956 and https://github.com/radian-software/straight.el/issues/955. Fixed in https://github.com/radian-software/straight.el/commit/b6dba13aa2660e4bff5e0bae9f7e684e0497b9ab.
